adapters/handlers/grpc/v1/service.go

```diff
@@ -341,14 +341,18 @@ func (s *Service) validateClassAndProperty(searchParams dto.GetParams) error {
 	return nil
 }
 
-func (s *Service) classGetterWithAuthzFunc(principal *models.Principal) func(string) (*models.Class, error) {
+func (s *Service) classGetterWithAuthzFunc(principal *models.Principal, tenant string) func(string) (*models.Class, error) {
 	authorizedCollections := map[string]*models.Class{}
 
 	return func(name string) (*models.Class, error) {
 		class, ok := authorizedCollections[name]
 		if !ok {
+			resources := authorization.CollectionsData(name)
+			if tenant != "" {
+				resources = authorization.ShardsData(name, tenant)
+			}
```

Comment:
```
		class, ok := authorizedCollections[name]
		if !ok {
```

this will not go in here if the collection was known before, eg if you have collection1/allowed_tenant and collection1/not_allowed_tenant the second one will also go through. However I am not sure if that input is possible here

Reply:
Ahhh, I see. Comparing it to the method in `batchObjects`, it needs that same logic here

Reply:
> It's used to get the class of not just the collection on which is being searched but also collections that are referenced in filters, where the tenant would not make sense

We do not want to repeatedly re-check permissions so we somehow need to "reuse" checks. 
Also, references can only be:
- to the same tenant for MT collections
- to non-MT colelctions (where only * makes sense for tenants)